server/src/com/mirth/connect/server/servlets/WebStartServlet.java

```diff
@@ -182,9 +182,16 @@ protected Document getAdministratorJnlp(HttpServletRequest request) throws Excep
         String serverHostname = request.getServerName();
         int serverPort = request.getServerPort();
         String contextPath = request.getContextPath();
-        String codebase = scheme + "://" + serverHostname + ":" + serverPort + contextPath;
 
         PropertiesConfiguration mirthProperties = getMirthProperties();
+        
+        String codebase = null;
+        if (StringUtils.isNotBlank(mirthProperties.getString("codebase.webstart.url"))) {

```

Comment:
PropertiesConfiguration supports[ a getString with a defaultValue argument ](https://commons.apache.org/proper/commons-configuration/apidocs/org/apache/commons/configuration2/AbstractConfiguration.html#getString(java.lang.String,java.lang.String)) 

Would using this getString method be better than an if/else condition? it is logically the same.

Reply:
I pushed the modification, I did the test and it works